Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/Point.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * a basic class that defines a proto-point
+ */
+@Getter
+@Setter
+public class Point {
+
+    // current values
+    double[] currentInput;
+
+    // input timestamp
+    long inputTimestamp;
+
+    public Point(double[] input, long inputTimestamp) {
+        this.currentInput = copyIfNotnull(input);
+        this.inputTimestamp = inputTimestamp;
+    }
+
+    public void setCurrentInput(double[] currentValues) {
```

Comment:
Does it make sense for this field to be settable? Is this method redundant with the `@Setter` annotation?

Reply:
We are stuck with the 2.1 state classes and the mappers at the moment. We can remove this as we bump up version (likely soon with more functionality in Impute)